### PR TITLE
Submit to a dashboard

### PR DESCRIPTION
**Pull request by vibraphone**
### Summary

This reworks the `test-build` script to use ctest in order to capture log information generated during configure, build, and test stages. The test stage just runs `attest` – eventually tests will get submitted individually to CDash though this will be a separate commit. Configure and build results are submitted to the [IBAMR CDash](https://open.cdash.org/index.php?project=IBAMR).

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -287,3 +287,32 @@ ENDIF()
 # Set up the default attest configuration file:
 CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/tests/attest.conf.in
   ${CMAKE_BINARY_DIR}/attest.conf)
+
+# For the time being, run `attest` as if it were a single test.
+set(disabled_tests)
+list(APPEND disabled_tests "mpirun=[5-9]")
```
Comment (drwells) [inline tests/CMakeLists.txt:293]:
```suggestion
list(APPEND disabled_tests "mpirun=[3-9]")
```
There are probably some other MPI settings we can tweak but for now we need to disable tests past 2 cores.

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -287,3 +287,32 @@ ENDIF()
...
+list(APPEND disabled_tests "explicit_ex1_2d.mpirun=4.input")
+list(APPEND disabled_tests "explicit_ex2_3d.nodal_quadrature.input")
+list(APPEND disabled_tests "explicit_ex4_3d.mpirun=4.input")
+list(APPEND disabled_tests "explicit_ex5_3d.mpirun=2.input")
+list(APPEND disabled_tests "explicit_ex8_2d.input")
+list(APPEND disabled_tests "explicit_ex8_2d.scratch_hier.input")
+list(APPEND disabled_tests "free_falling_cyl_cib|cib_plate")
+list(APPEND disabled_tests "nwt_cylinder|rotating_barge|cib_double_shell")
+list(APPEND disabled_tests "restart")
```
Comment (drwells) [inline tests/CMakeLists.txt:302]: I think the restart tests are fine (as long as they don't use more than 2 cores) but we can worry about this later.

File: attest [hunk 3]
```diff
@@ -465,7 +465,7 @@ def main():
                         "a test is in 'tests/dir/example.input', the regex "
                         "will be applied to 'dir/example.input'.")
     parser.add_argument('--test-timeout', type=int,
-                        default=conf['test_timeout'],
+                        default=conf['test_timeout'] if 'test_timeout' in conf else 1200,
```
Comment (drwells) [inline attest:468]:
```suggestion
                        default=conf['test_timeout'],
```
Fixed by #1289 - you may need to rebase.

**Review by drwells — CHANGES_REQUESTED** [review]

File: tests/CMakeLists.txt (hunk 2 above)
Comment (drwells) [reply tests/CMakeLists.txt:302]: The restart tests are fine - we just need the MPI limit.

File: attest (hunk 3 above)
Comment (vibraphone) [reply attest]: Done.

File: tests/CMakeLists.txt (hunk 1 above)
Comment (vibraphone) [reply tests/CMakeLists.txt]: Done

**Review by drwells — APPROVED** [review]
This is a significant step forward so lets merge - the CI is able to finish successfully and we are logging the results.

**Pull request merged**